addons/hr_contract/migrations/13.0.1.0/post-migration.py

```diff
@@ -0,0 +1,35 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+_unlink_by_xmlid = [
+    # hr.contract.type
+    'hr_contract.hr_contract_type_emp',
+    'hr_contract.hr_contract_type_sub',
+    'hr_contract.hr_contract_type_wrkr',
+]
+
+
+def fill_contract_company_id(cr):
+    openupgrade.logged_query(
+        cr, """
+        UPDATE hr_contract hc
+        SET company_id = c.id
+        FROM (SELECT id FROM res_company ORDER BY id LIMIT 1) AS c
+        WHERE hc.company_id IS NULL
+        """
+    )
+
+
+def fill_employee_contract_id(env):
+    contract = env['hr.contract']
+    employees = env['hr.employee'].search([])
+    for employee in employees:
+        employee.contract_id = contract.search([('employee_id', '=', employee.id)], order='date_start desc', limit=1)
```

Comment:
```suggestion
        employee.contract_id = contract.search([('employee_id', '=', employee.id), ('company_id', '=', employee.company_id.id)], order='date_start desc', limit=1)
```
cannot be done using sql queries? if not, then do read_group or alike, because a `search` in a `for` is not good.